app/parsers/markdown-parser.rb

```diff
@@ -0,0 +1,166 @@
+require 'redcarpet'
+
+# TODO: buradan asagidaki fonksiyonlar standartlar belirlenerek duzenlenecek
+# TODO: ic ice döngüler sırasında parse işlemi sorun vermekte. Düzeltilmesi gerekiyor
+
+# öyak çeviri yönetim sistemi için kullanılabilecek standartlarda render
+class OyakRender < Redcarpet::Render::HTML
+  # header tagları icin
+  def header(text, header_level)
+    %(<header#{header_level}>#{text}</header#{header_level}>)
+  end
+
+  # paragraf olarak adlandırılan düz yazıların ayar kısmı
+  def paragraph(text)
+    %(<text>#{text}</text>)
+  end
+
+  def block_quote(quote)
+    %(<blockquote>#{quote}</blockquote>)
+  end
+
+  def footnotes(content)
+    %(<footnotes>#{content}</footnotes>)
+  end
+
+  # list name not working yet
+  def list(contents, _list_type)
+    "<list>#{contents}</list>"
+  end
+
+  def list_item(text, _list_type)
+    "<list_item>#{text}</list_item>"
+  end
+
+  # table name not working yet
+  def table(header, body)
+    "<table>#{header}</table> #{body}"
+  end
+
+  def table_row(content)
+    "<tableRow>#{content}</tableRow>"
+  end
+
+  def table_cell(content, alignment)
+    "<tableCell>#{alignment} #{content}</tableCell>"
+  end
+
+  def image(_link, title, _alt_text)
+    %(<resim>[resim: #{title}]</resim>)
+  end
+
+  def link(_link, title, _content)
+    %(<link>#{title}</link>)
+  end
+
+  def raw_html(_raw_html)
+    %(raw_tml)
+  end
+
+  def linebreak
+    "\n"
+  end
+
+  def triple_emphasis(text)
+    text
+  end
+
+  def strikethrough(text)
+    text
+  end
+
+  def superscript(text)
+    text
+  end
+
+  def underline(text)
+    text
+  end
+
+  def highlight(text)
+    text
+  end
+
+  def quote(text)
+    text
+  end
+
+  def footnote_ref(number)
+    number
+  end
+
+  def emphasis(text)
+    text
+  end
+
+  def double_emphasis(text)
+    text
+  end
+
+  def autolink(link, _link_type)
+    link
+  end
+
+  def codespan(code)
+    code
+  end
+end
+
+# basit bir makaleyi parcalayıp kullanılabilir dizi haline getirir
+class MarkdownParser
+  # verilen markdown stringini diziye parcalar
+  def self.parse2arr(markdown_string, options = {}, render = OyakRender)
+    md = Redcarpet::Markdown.new(render, options)
+    result_html = md.render(markdown_string)
+    result = result_html.split(/<\/[^>]*>/)
+    result.inject([]) { |arr, v| arr << v.scan(/<([^>]*)>(.*)/).first }
```

Comment:
Sanırım işin can alıcı kısımları buralar diyebiliriz :smile:  Bu önemli ve anlaşılması zor yerleri yorum satırlarıyla desteklersek güzel olur. :+1: